### PR TITLE
Initial Proposal for a Phonebank profile

### PR DESCRIPTION
**Pull request by augfrank**

### CODE REVIEW

File: phone_bank.md [hunk 1]
```diff
@@ -0,0 +1,552 @@
...
+{% include global_fields.md %}
+
+_[Back to top...](#)_
+
+### Phonebank Fields
+
+| Name          | Type                | Description
+| -----------   | -----------         | --------------
+|origin_system      |string     |A human readable identifier of the system where this Phonebank was created. (ex: "OSDI System")
+|name               |string     |The name of the Phonebank.
```
Comment (j-ro) [inline phone_bank.md]: sync description with name in other places (see canvassing effort) and add title?

Comment (j-ro) [reply phone_bank.md]: also summary?

File: phone_bank.md [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+|description        |string     |A description of the Phonebank, usually displayed publicly. May contain text and/or HTML.
+|start_time        |datetime     |The start date and time for the Phone banking Effort.
+|end_time        |datetime     |The end date and time for the Phone banking Effort.
+|daily_start_hour |integer     |The hour of the day volunteers should start calling.
```
Comment (j-ro) [inline phone_bank.md]: clarify that this is 24 hour time? or is that implied via the integer type?

File: phone_bank.md [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+|daily_stop_hour |integer     |The hour of the day volunteers should stop calling.
+|caller_id        |string     |The phone number that is displayed as the caller-id for this phonebank
```
Comment (j-ro) [inline phone_bank.md]: should this be a string or an integer?

File: phone_bank.md [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+|status           |enum      |Status of the Phonebank. Possible values are: "draft", "pending", "scheduled", "started", or "stopped"
```
Comment (j-ro) [inline phone_bank.md]: what does pending mean?

Comment (j-ro) [reply phone_bank.md]: and maybe this is flexenum, so another value can be used?

File: phone_bank.md [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+|dialing_mode           |enum      |The type of the dialing for this phonebank. One of "predictive", "power", "manual", or another value.
```
Comment (j-ro) [inline phone_bank.md]: type is flexenum, right?

File: phone_bank.md [hunk 6]
```diff
@@ -0,0 +1,552 @@
...
+|voicemail_audio_url    |string      |A URL string pointing to a audio file used for voice mail audio drops.
+|administrative_url    |string      |A URL string pointing to the Phonebank administrative page on the web.
+|browser_url    |string      |A URL string pointing to the Phonebank public page on the web.
+|total_completed	|integer	|A read-only computed property representing the current count of the total number completed on the Phonebank.
```
Comment (j-ro) [inline phone_bank.md]: probably should be "total_responses" to go with the naming style in other places

File: phone_bank.md [hunk 7]
```diff
@@ -0,0 +1,552 @@
...
+
+{% include links_intro.md %}
+
+| Name          | Type       | Description
+|-----------    |----------- |-----------
+|self           |[Phonebank*](phone_bank.html)    |A self-referential link to the phone bank.
+|creator        |[Person*](people.html)         |A link to a single Person resource representing the creator of the Phonebank.
+|modified_by    |[Person* ](people.html)        |A link to a Person resource representing the last editor of this Phonebank.
+|taggings       |[Taggings[]* ](people.html)    |A link to the collection of Tagging resources for this Phonebank.
+|phone_bank_result      |[Phonebank_result[]*](phone_bank_result.html)        |A link to a collection of Phone bank results resources representing the results that occured.
```
Comment (j-ro) [inline phone_bank.md]: name and type should be plural

File: phone_bank.md [hunk 8]
```diff
@@ -0,0 +1,552 @@
...
+|script  |[Script*](scripts.html) | A link to the Script resource associated with this Phonebank.
+|target_list  |[List*](lists.html) | A link to the List resource that represents the list of People who will receive calls in this phonebank.
+|advocacy_target  |[List*](lists.html) | A link to the List resource that represents the list of People who will be advocacy targets in this phonebank.
```
Comment (j-ro) [inline phone_bank.md]: let's discuss this one a bit more -- I think I understand it, but so far we haven't really represented advocacy targets as separate resources. See advocacy campaigns.

File: phone_bank.md [hunk 9]
```diff
@@ -0,0 +1,552 @@
...
+|volunteers  |[List*](lists.html) | A link to the List resource that represents the list of People who will be the volunteers in this phonebank.
+|events  |[List*](lists.html) | A link to the List resource that represents a collection of events associated with this Phonebank.
```
Comment (j-ro) [inline phone_bank.md]: I think I understand this, but maybe the description can be more clear about what associating an event with a phonebank actually means? Is it the event where volunteers gather to phonebank? or events they're signing people they're calling up for?

File: phone_bank.md [hunk 10]
```diff
@@ -0,0 +1,552 @@
...
+
+
+_[Back to top...](#)_
+
+
+## Related Resources
+
+* [Person](people.html)
+* [Script](scripts.html)
+* [Phone bank result](phone_bank_result.html)
```
Comment (j-ro) [inline phone_bank.md]: add List, Taggings, basically anything that's linked to above

File: phone_bank.md [hunk 11]
```diff
@@ -0,0 +1,552 @@
...
+                    "osdi:target_list": {
+                        "href": "https://osdi-sample-system.org/api/v1/lists/1efc3644-af25-4253-90b8-a0baf12dbd1e"
+                    },
+                    "osdi:advocacy_target": {
+                        "href": "https://osdi-sample-system.org/api/v1/lists/d91b4b2e-ae0e-4cd3-9ed7-d0ec501b0bc3"
+                    },
+                    "osdi:volunteers": {
+                        "href": "https://osdi-sample-system.org/api/v1/lists/65345d7d-cd24-466a-a698-4a7686ef684f"
+                    },
+                    "osdi:events": {
```
Comment (j-ro) [inline phone_bank.md:205]: would this be linking to a list? I guess a list of events?

File: phone_bank.md [hunk 12]
```diff
@@ -0,0 +1,552 @@
...
+                        "href": "https://osdi-sample-system.org/api/v1/lists/1efc3644-af25-4253-90b8-a0baf12dbd1e"
+                    },
+                    "osdi:phone_bank_result": {
+                        "href": "https://osdi-sample-system.org/api/v1/phone_bank/a91b4b2e-ae0e-4cd3-9ed7-d0ec501b0bca/phone_bank_result"
```
Comment (j-ro) [inline phone_bank.md]: phone_bank_results, plural, right?

File: phone_bank_result.md [hunk 13]
```diff
@@ -0,0 +1,804 @@
...
+    * [Scenario: Retrieving Answers for an individual Phonebank result resource (GET)](#scenario-retrieving-answers-for-an-individual-phone-bank-result-get)
+    * [Scenario: Retrieving Taggings for an individual Phonebank result resource (GET)](#scenario-retrieving-taggings-for-an-individual-phone-bank-result-get)
+    * [Scenario: Creating a new Phonebank result (POST)](#scenario-creating-a-new-phone-bank-result-post)
+    * [Scenario: Modifying a Phonebank result (PUT)](#scenario-modifying-a-phone-bank-result-put)
+    * [Scenario: Deleting a Phonebank result (DELETE)](#scenario-deleting-a-phone-bank-result-delete)
+
+
+{% include endpoints_and_url_structures.md %}
+
+The link relation label for an Phone bank result resource is ```osdi:phone-bank-result``` for a single resource or ```osdi:phone-bank-results``` for a collection of resources.
```
Comment (j-ro) [inline phone_bank_result.md]: underscore here instead of dash

File: phone_bank_result.md [hunk 14]
```diff
@@ -0,0 +1,804 @@
...
+_[Back to top...](#)_
+
+
+### Phone bank result Fields
+
+| Name          | Type      | Description
+|-----------    |-----------|--------------
+|origin_system	  |string    |A human readable identifier of the system where this Phonebank result was created. (ex: "OSDI System")
+|call_date		  |string	 |The date and time the call was made.
+|caller_id        |string    |The phone number that is displayed as the caller-id for this call
```
Comment (j-ro) [inline phone_bank_result.md:53]: integer type?

File: phone_bank_result.md [hunk 15]
```diff
@@ -0,0 +1,804 @@
...
+|phone_number     |string    |The phone number that was called
```
Comment (j-ro) [inline phone_bank_result.md:54]: integer type?

File: phone_bank_result.md [hunk 16]
```diff
@@ -0,0 +1,804 @@
...
+|duration         |string    |The length of the phone call in seconds. Calls that do not connect will have zero in this field.
```
Comment (j-ro) [inline phone_bank_result.md]: integer type?

File: phone_bank_result.md [hunk 17]
```diff
@@ -0,0 +1,804 @@
...
+|disposition      |enum      |Disposition of the phone call. Possible values are "answer". "busy", "do-not-call", "meaningful interaction", "machine" etc..
```
Comment (j-ro) [inline phone_bank_result.md]: probably should be called status_code, see canvasses, and maybe add the success bool too?

File: phone_bank_result.md [hunk 18]
```diff
@@ -0,0 +1,804 @@
...
+
+{% include links_intro.md %}
+
+|Name          	|Type		|Description
+|-----------    |-----------|--------------
+|self			|[Phonebank Result*](phone-bank-result.html)	|A self-referential link to the Phonebank result.
+|caller         |[Person*](people.html)         |A link to a single Person resource representing the person who made the call
+|target		    |[Person*](people.html)  |A link to a single Person resource representing the person who was contacted.
+|answers        |[Answers[]*](answers.html) | A link to the collection of Answers to [Questions](questions.html) posed during the Phonebank.
+|attendances    |[Attendances[]*](attendances.html) | A link to the collection of Attendances to [Events](events.html) posed during the Phonebank.
```
Comment (j-ro) [inline phone_bank_result.md]: let's clarify the description a bit maybe -- I get what it's going for, just could be more clear

File: phone_bank_result.md [hunk 19]
```diff
@@ -0,0 +1,804 @@
...
+
+## Related Resources
+
+* [Question](questions.html)
+* [Person](people.html)
+* [Answer](answers.html)
+* [Tagging](taggings.html)
+* [Phonebank](phone_bank.html)
+* [Attendances](attendances.html)
+* [Events](events.html)
```
Comment (j-ro) [inline phone_bank_result.md]: remove, not directly related

File: phone_bank_result.md [hunk 20]
```diff
@@ -0,0 +1,804 @@
...
+    "per_page": 25,
+    "page": 1,
+    "total_records": 250,
+    "_links": {
+        "next": {
+            "href": "https://osdi-sample-system.org/api/v1/people/c945d6fe-929e-11e3-a2e9-12313d316c29/phone-bank-result?page=2"
+        },
+        "osdi:phone-bank-results": [
+            {
+                "href": "https://osdi-sample-system.org/api/v1/people/c945d6fe-929e-11e3-a2e9-12313d316c29/phone-bank-result/d91b4b2e-ae0e-4cd3-9ed7-d0ec501b0bc3"
```
Comment (j-ro) [inline phone_bank_result.md]: underscore and plural throughout

File: phone_bank_result.md [hunk 21]
```diff
@@ -0,0 +1,804 @@
...
+    "per_page": 25,
+    "page": 1,
+    "total_records": 250,
+    "_links": {
+        "next": {
+            "href": "https://osdi-sample-system.org/api/v1/phone_bank/d945d6fe-929e-11e3-a2e9-12313d316c29/phone-bank-result?page=2"
+        },
+        "osdi:phone-bank-results": [
+            {
+                "href": "https://osdi-sample-system.org/api/v1/people/c945d6fe-929e-11e3-a2e9-12313d316c29/phone-bank-result/d91b4b2e-ae0e-4cd3-9ed7-d0ec501b0bc3"
```
Comment (j-ro) [inline phone_bank_result.md]: shouldn't link to people here, right?

Comment (j-ro) [reply phone_bank_result.md]: this is also wrong on canvasses, should be fixed...

**Review by j-ro — COMMENTED** [review]
We also need a helper doc

File: phone_bank_result.md (hunk 21 above)
Comment (joshco) [reply phone_bank_result.md]: @augfrank we are waiting on you to integrate Jason's comments, then we will approve

**Pull request merged**